src/sinks/splunk_hec/logs/sink.rs

```diff
@@ -150,17 +153,31 @@ pub fn process_log(
 
     let host = log.get(host_key).cloned();
 
-    let timestamp = match log.remove(log_schema().timestamp_key()) {
-        Some(Value::Timestamp(ts)) => ts,
-        _ => chrono::Utc::now(),
+    let timestamp = if timestamp_key.is_empty() {
```

Comment:
Nit: I think you can use `timestamp_key.map(...)` here.

Reply:
As discussed, dropping this since the timestamp_key is a &str and you'd thought it was an Option<&str> when making this comment.